### PR TITLE
Promote paid contact features

### PR DESCRIPTION
**Pull request by EpokK**
Block edition for non paid mail account.
Capture d’écran 2019-07-25 à 15 28 19
Capture d’écran 2019-07-25 à 15 29 30

### CODE REVIEW

File: src/app/components/ContactView.js [hunk 1]
```diff
@@ -44,17 +45,16 @@ const ContactView = ({ properties, contactID, errors }) => {
...
-                        <ContactViewProperties contactID={contactID} properties={properties} field="tel" />
-                        <ContactViewProperties contactID={contactID} properties={properties} field="adr" />
-                        <ContactViewProperties contactID={contactID} properties={properties} />
-                    </>
-                ) : (
-                    <div className="mb1 pl1 pr1">
-                        <ContactPromote />
-                    </div>
-                )}
+                <div className="relative">
```
Comment (econdepe) [inline src/app/components/ContactView.js]: Doing things this way, what if a free user edits the HTML code and simply erases the
```html
<div className="contactView-locked-background">
    <ContactPromote />
</div>
```
?
Will she have access to the locked properties? Or are these properties actually empty for a free user?

File: src/app/components/ContactView.scss [hunk 2]
```diff
@@ -0,0 +1,24 @@
+@import "~design-system/_sass/reusable-components/design-system-config";
+
+.contactView-locked-background {
+    position: absolute;
+    display: flex;
+    align-items: center;
+    justify-content: center;
+    z-index: 2;
```
Comment (econdepe) [inline src/app/components/ContactView.scss]: are we sure this z-index is high enough? No possibility of other elements having a higher one?

File: src/app/components/ContactPromote.js [hunk 3]
```diff
@@ -1,18 +1,21 @@
...
             </h3>
             <Alert learnMore="https://protonmail.com/support/knowledge-base/encrypted-contacts/">{c('Info')
                 .t`Upgrade your plan to unlock encrypted details such as phone numbers and home addresses.`}</Alert>
-            <a href="/settings/dashboard" className="pm-button pm-button pm-button--primary">{c('Action')
-                .t`Upgrade`}</a>
-        </>
+            <div className="mb1">
+                <Href url="/settings/subscription" target="_self" className="pm-button pm-button pm-button--primary">{c(
+                    'Action'
+                ).t`Upgrade`}</Href>
```
Comment (Yiin) [inline src/app/components/ContactPromote.js]:
```
<Href ...>
   {c('Action').t`Upgrade`}
</Href>
```
would look a bit better imo.

File: src/app/components/ContactView.scss [hunk 4]
```diff
@@ -0,0 +1,28 @@
...
+    left: 0;
+    width: 100%;
+    height: 100%;
+    padding: 2em;
+    background: rgba($white, .3);
+}
+
+.contactView-locked-background ~ * {
+    pointer-events: none;
+    filter:blur(1px);
```
Comment (Yiin) [inline src/app/components/ContactView.scss]: 😱 missing a space after `:`

File: src/app/components/ContactPromote.js [hunk 5]
```diff
@@ -1,18 +1,21 @@
...
             <h3 className="mb1">
                 <Icon name="lock" /> {c('Title').t`Encrypted details`}
             </h3>
             <Alert learnMore="https://protonmail.com/support/knowledge-base/encrypted-contacts/">{c('Info')
                 .t`Upgrade your plan to unlock encrypted details such as phone numbers and home addresses.`}</Alert>
-            <a href="/settings/dashboard" className="pm-button pm-button pm-button--primary">{c('Action')
-                .t`Upgrade`}</a>
-        </>
+            <div className="mb1">
+                <Href url="/settings/subscription" target="_self" className="pm-button pm-button pm-button--primary">{c(
```
Comment (Yiin) [inline src/app/components/ContactPromote.js]: `pm-button` written twice

File: src/app/components/ContactView.js (hunk 1 above)
Comment (EpokK) [reply src/app/components/ContactView.js]: That's OK.

**Pull request merged**